Installation on Alpine Linux

Wrong-Code opened the pull request:
I have extended the Technitium DNS installation script to properly install on Alpine Linux. The script now takes care of installing the binaries needed just for the installation, as well as the libstdc++ package required by Microsoft .NET (the dotnet installation scripts does not check if it is missing)

I also provide the `openrc.service` file to run Technitium DNS as a service. The installation script now checks not only for systemd, but also for OpenRC.

If this PR is accepted, [this blog post](https://blog.technitium.com/2017/11/running-dns-server-on-ubuntu-linux.html) should be updated stating the now supported Alpine Linux.

I have tested the script on Alpine Linux v3.20.

Wrong-Code committed:
Initial work to support installation on Alpine Linux

Wrong-Code committed:
Add support for installation on Alpine Linux

Wrong-Code committed:
Fix: if no virtual packages were installed, let 'apk del ...' fail silently.

Wrong-Code commented:
Hi,

I have rebased my PR with release 14.3. It still works on Alpine even with .NET 9, I hope you will have time to integrate it.

ShreyasZare commented:
Thanks. I have this PR starred in my inbox so will get it review once I have sufficient time to do so.

EliteSalman committed:
Update DNS server image reference to pull podman from Docker Hub (#1668)

ShreyasZare committed:
install.sh: updated install script to correctly configure network manager so that it does not alter resolv.conf. Also made changes to allow restoring original resolv.conf on uninstall.

ShreyasZare committed:
uninstall.sh: updated script to restore network manager to default config, added support to restore original resolv.conf file along with starting up systemd-resolved. Also implemented option to remove config folder after uninstall.

ShreyasZare committed:
added contributing.md file

ShreyasZare committed:
updated security.md file

ShreyasZare committed:
added SupportedRFCs.md file

ShreyasZare committed:
install.sh: updated install script to run dns service as non-root user.

ShreyasZare committed:
uninstall.sh: updated script to support install script changes. Added option to remove config and logs files on disk after uninstall.

ShreyasZare committed:
systemd.service: updated systemd service config to run as non-root user with additional hardening.

ShreyasZare committed:
DnsServerApp: updated app to allow running explicitly as a portable app.

ShreyasZare committed:
AuthManager: Updated SaveConfigFileInternal() to write config in tmp file before replacing it. Updated implementation to support new ClusterApiToken type. Code refactoring changes done.

ShreyasZare committed:
UserSession: updated implementation to support ClusterApiToken and SingleUse session types. Added UpgradeToClusterApiToken() for backward compatibility usage. Updated HasExpired() to handle new session types and fix minor bug.

ShreyasZare committed:
ClusterManager: Updated constructor to find and convert API token to Cluster API Token. Updated SaveConfigFileInternal() to save config in tmp file before moving. Updated code to support new Cluter API Token. Code refactoring changes and minor fixes done.

ShreyasZare committed:
ClusterNode: Updated GetApiClient() to support cluster API token. Code refactoring changes done.

ShreyasZare committed:
ClientFullyQualifiedDomainNameOption: updated ParseOptionValue() to handle case of buggy client implementation.

ShreyasZare committed:
DhcpServer: Updated DNS entry management implementation to allow having persistent DNS records for reserved leases with hostname configured even when reserved lease was not allocated. Updated SaveScopeFile() to write config to tmp before moving it. Code refactoring changes done.

ShreyasZare committed:
Scope: Updated implementation to support persistent DNS entries for reserved leases with hostname configured even when the lease was not allocated. Code refactoring changes done.

ShreyasZare committed:
DnsApplication: Updated SetConfigAsync() to write config to tmp before moving it.

ShreyasZare committed:
DnsApplicationManager: code refactoring changes done.

ShreyasZare committed:
InternalDnsServer: code refactoring changes done.

ShreyasZare committed:
AllowedZoneManager: Updated SaveZoneFileInternal() to write config to tmp before moving. Code refactoring changes done.

ShreyasZare committed:
BlockedZoneManager: Updated SaveZoneFileInternal() to write config to tmp before moving. Code refactoring changes done.

ShreyasZare committed:
BlockListZoneManager: Updated SaveConfigFileInternal() to write config to tmp before moving. Code refactoring changes done.

ShreyasZare committed:
AuthZoneManager: Updated SaveZoneFileInternal() to write config to tmp before moving. Added ActivatePrimaryZoneKskDnsKey() method. Updated InternalQuery() to support RFC 7314 EDNS EXPIRE option. Code refactoring and other minor changes done.

ShreyasZare committed:
CacheZoneManager: Updated ResolveCNAME() and Query() to allow querying for delegation NS record. Code refactoring changes done.

ShreyasZare committed:
ApexZone: code refactoring changes done.

ShreyasZare committed:
PrimaryZone: implemented ActivateKskDnsKey() method. Code refactoring changes done.

ShreyasZare committed:
SecondaryZone: Implemented RFC 7314 EDNS EXPIRE option support. Code refactoring changes done.

ShreyasZare committed:
StubZone: Code refactoring changes done.

ShreyasZare committed:
DnsServer: Replaced PreferIPv6 with IPv6Mode feature. Implemented new EDNS Client Subnet Source Address feature. Added new env vars for inmemory stats and max stats file days options. Updated SaveConfigFileInternal() to write config to tmp before moving. Implemented new feature to allow dynamic binding for UDP listeners on first request when using ANY local ep addresses to allow sending response to the correct interface. Fixed bug in ProcessConnectionAsync() for missing proxy protocol IsLocal check. Fixed implementation issues for QUIC listener in AcceptQuicConnectionAsync(). Updated PostProcessQueryAsync() to prevent adding EDNS OPT for zone transfer responses. Implemented RFC 7314 EDNS EXPIRE option support. Fixed issue  in AuthoritativeQueryAsync() that caused APP record to be ignored in certain cases. Updated RecursiveResolverBackgroundTaskAsync() to measure correct response time. Updated DefaultRecursiveResolveAsync() to add missing RCODE check for response before acccepting it and updated to measure correct response time. Updated PrefetchCacheAsync() to add missing ECS options. Updated auto cache prefetch feature implementation to fix multiple issues. Code refactoring and multiple minor fixes done.

ShreyasZare committed:
ResolverDnsCache: Code refactoring changes done.

ShreyasZare committed:
StatsManager: implemented support for lifetime counters.

ShreyasZare committed:
DnsWebService: Added explicit portable app support. Updated SaveConfigFileInternal() to write config to tmp before moving. Added API support for single use token, metrics, and option to manually activate KSK. Updated WebServiceExceptionHandler() to remove stack trace from response. Updated API implementation to support Authorization Bearer token. Updated TryGetSession() to support single use token. Updated GetSessionUser() to support cluster api token. Code refactoring changes done.

ShreyasZare committed:
DnsWebServiceLegacy: code refactoring changes done.

ShreyasZare committed:
LogManager: updated implementation to use platform specific log folder path when not running as a portable app. Added new env vars for log folder path and max log file days. Updated SaveConfigFileInternal() to write config to tmp before moving. Updated StopLoggingAsync() to wait for pending logs to be written before stopping.  Code refactoring changes done.

ShreyasZare committed:
WebServiceApi: Updated CheckForUpdateAsync() to remove extra data from being logged to avoid user confusion. Code refactoring changes done.

ShreyasZare committed:
WebServiceAuthApi: updated implementation to support cluster api token and single use tokens. Updated SetUserDetaails() to check for cluster user to prevent disabling it. Code refactoring changes done.

ShreyasZare committed:
WebServiceDhcpApi: code refactoring changes done.

ShreyasZare committed:
WebServiceSettingsApi: Updated RestartService() to change restart sequence to prevent web service lockout. Updated API for ipv6 mode and ECS source address support. Code refactoring changes done.

ShreyasZare committed:
WebServiceZonesApi: Updated code to support NS misconfigured mark feature. Updated ReadRecordsToImportFromAsync() to set last modified property for records. Updated UnsignPrimaryZone() to prevent unsigining of cluster primary zone. Added ActivatePrimaryZoneKskDnsKey() to allow manually activating KSK. Code refactoring and other minor changes done.

ShreyasZare committed:
IDnsServer: Updated interface with a breaking change to support IPv6 Mode feature.

ShreyasZare committed:
HttpApiClient: Updated constructors to support ipv6 mode and setting http timeout.

ShreyasZare committed:
DnsServiceWorker: code refactoring changes done.

ShreyasZare committed:
inno: updated appinstall code to allow new installations to run service as a non-system user.

ShreyasZare committed:
inno: updated service code to support configuring LogOnAs parameter.

ShreyasZare committed:
docker-compose: Updated file to add new env vars. Updated to support platform specific log path which mounts to a separate volume.

ShreyasZare committed:
updated docker env vars to add 4 new variables.

ShreyasZare committed:
Merge branch 'develop' of https://github.com/TechnitiumSoftware/DnsServer into develop

ShreyasZare committed:
main.css: added new container width.

ShreyasZare committed:
fixed opendns addresses in dnsclient server list

ShreyasZare committed:
fixed opendns addresses in quick forwarders list

ShreyasZare committed:
webapp: updated html to support new dashboard blocking options. Updated to support new IPv6 mode feature. Updated to support EDNS Client Subnet (ECS) Source Address feature. Updated to support live update feature in Query Logs. Other minor changes done.

ShreyasZare committed:
common.js: updated HTTPRequest() to support authorization bearer tokens. Code refactoring changes done.

ShreyasZare committed:
apps.js: implemented bearer token support.

ShreyasZare committed:
auth.js: implemented bearer token support. Added support for cluster api token.

ShreyasZare committed:
cluster.js: implemented bearer token support. Other minor changes done.

ShreyasZare committed:
dhcp.js: implemented bearer token support.

ShreyasZare committed:
dnsclient.js: implemented bearer token support.

ShreyasZare committed:
logs.js: implemented bearer token support. Implemented live update feature for Query Logs. Other minor changes done.

ShreyasZare committed:
other-zones.js: implemented bearer token support.

ShreyasZare committed:
zone.js: implemented bearer token support. Added option to manually activate KSK.

ShreyasZare committed:
main.js: implemented bearer token support. Updated code to support ECS source address feature and ipv6 mode. Added support for dashboard blocking options.

ShreyasZare committed:
AdvancedBlocking: code refactoring changes done.

ShreyasZare committed:
AdvancedBlocking: updated config example to avoid user confusion.

ShreyasZare committed:
AutoPtr: minor optimization changes.

ShreyasZare committed:
BlockPage: updated app to implement online cert signing feature to allow SSL MiTM when app's root cert is installed on clients. implemented nullable. Code refactoring changes done.

ShreyasZare committed:
Dns64: minor changes done.

ShreyasZare committed:
DnsBlockList: code refactoring changes done.

ShreyasZare committed:
Failover: code refactoring changes done.

ShreyasZare committed:
QueryLogsMySql: updated code to log TC response correctly.

ShreyasZare committed:
QueryLogsSqlite: updated code to log TC response correctly.

ShreyasZare committed:
QueryLogsSqlServer: updated code to log TC response correctly.

ShreyasZare committed:
SplitHorizon.AddressTranslation: minor code changes.

ShreyasZare committed:
WildIp: updated app implementation to improve parsing logic. Added new option to allow configuring allowed networks to prevent misuse/abuse.

ShreyasZare committed:
webapp: minor css changes.

ShreyasZare committed:
install.sh: updated dotnet to 10.0

ShreyasZare committed:
systemd.service: added env var to allow RSA+SHA1 DNSSEC algo.

ShreyasZare committed:
code refactoring changes done.

ShreyasZare committed:
ClusterManager: Updated UpdateClusterCatalogZoneOptions() to fix missing address family check.

ShreyasZare committed:
DnsServer: Added new option to configure udp listener socket send and receive buffer size.

ShreyasZare committed:
WebServiceSettingsApi: added support for udp send and receive buffer size options.

ShreyasZare committed:
webapp: Implemented support for new theme options. Added support for udp send/recv buffer size options.

ShreyasZare committed:
main.js: Implemented new theme options feature. Added support for udp send/recv buffer size options.

ShreyasZare committed:
logs.js: fixed minor issue of resetting entries per page value.

ShreyasZare committed:
main.css: updated css for new theme options implementation.

Hezer committed:
Fix dark mode readability and input elements using light theme (#1637)

* Adjust the contrast of colored elements, fix code and input elements in dark mode

* Change also the Error label color

ShreyasZare committed:
main.css: fixed minor issue with close in dark mode PR.

ShreyasZare committed:
AuthManager: updated implementation to support SSO. Code refactoring changes done.

ShreyasZare committed:
User: updated implementation to support SSO. Code refactoring changes done.

ShreyasZare committed:
ClusterManager: Updated InitializeCluster() to disallow sso user to init cluster. Code refactoring changes done.

ShreyasZare committed:
ClusterNode: added CreateSsoUserAsync() and SetSsoUserAsync() api methods.

ShreyasZare committed:
DnsWebService: Updated code to implement SSO with OIDC. Moved RestartService() method here. Code refactoring changes done.

ShreyasZare committed:
Extensions: added new TryQueryOrForm() and TryQueryOrFormArray() methods. Code refactoring changes done.

ShreyasZare committed:
WebServiceAuthApi: updated code to implement SSO. Updated the Create Token API to allow creating new token using the standard user session for authentication. Code refactoring changes done.

ShreyasZare committed:
WebServiceClusterApi: code refactoring changes done.

ShreyasZare committed:
WebServiceSettingsApi: moved RestartService() to main class. Code refactoring changes done.

ShreyasZare committed:
HttpApiClient: Updated implementation to use Authorization Bearer header instead of token param. Added CreateSsoUserAsync() and SetSsoUserAsync() to support SSO. Code refactoring changes done.

ShreyasZare committed:
updated docker env vars for SSO variables.

ShreyasZare committed:
webapp: added oidc logo image.

ShreyasZare committed:
auth.js: implemented code for SSO support. Implemented create token API changes. Code refactoring and other minor changes done.

ShreyasZare committed:
common.js: added getCookie() and setCookie() methods.

ShreyasZare committed:
main.js: updated code to support SSO. Other minor changes done.

ShreyasZare committed:
webapp: updated html to support SSO. Other minor changes done.

ShreyasZare committed:
DomainTree: removed "-" validation check.

ShreyasZare committed:
AuthZoneManager: Added DeleteAllRecords() method to delete all records except SOA and DNSSEC records. Updated ImportRecords() to add new zone overwrite option. Code refactoring changes done.

ShreyasZare committed:
BlockListZoneManager: Updated BlockListUrls property to add URL validation.

ShreyasZare committed:
ApexZone: Updated code to use InvariantCulture.

ShreyasZare committed:
DnsServer: Updated ProcessCNAMEAsync() to set ServerFailure RCODE and EDE in response when CNAME loop is detected. Fixed missing check for "this-server" in PriorityConditionalForwarderResolveAsync().

ShreyasZare committed:
StatsManager: Updated code to use InvariantCulture.

ShreyasZare committed:
LogManager: Updated code to use InvariantCulture.

ShreyasZare committed:
DnsWebService: Updated StartWebService() to set additional security headers. Updated ValidateQuicSupport() to check for IPv6 support and give specific error message.

ShreyasZare committed:
WebServiceApi: code refactoring changes.

ShreyasZare committed:
WebServiceClusterApi: updated code to use InvariantCulture.

ShreyasZare committed:
WebServiceLogsApi: updated code to use InvariantCulture.

ShreyasZare committed:
WebServiceSettingsApi: updated code to use InvariantCulture.

ShreyasZare committed:
WebServiceZonesApi: Updated ImportZoneAsync() to add new "overwriteZone" option.

ShreyasZare committed:
zone.js: updated code to add support for import zone overwite zone option.

ShreyasZare committed:
webapp: updated html to support overwrite zone option for import zone feature. Minor documentation changes done.

ShreyasZare committed:
minor changes.

ShreyasZare committed:
added Quad9 DoQ entries to json files.

ShreyasZare committed:
DnsServerCore.ApplicationCommon: Updated interfaces to support nullable.

ShreyasZare committed:
DnsServerSystemTrayApp: nullable related changes done.

ShreyasZare committed:
DnsServiceWorker: nullable related changes done.

ShreyasZare committed:
updated supporteed RFCs

ShreyasZare committed:
updated API docs

lachlan-stevens committed:
Fix incorrect endpoint for UpdateIpAddress in docs (#1830)

ShreyasZare committed:
AdvancedBlocking: updated code to support comments in config.

ShreyasZare committed:
Updated apps to support conmments in config and APP record data.

ShreyasZare committed:
DnsRebindingProtection: Updated code to support comments in config. Updated PostProcessAsync() to bypass checks for blocked responses.

ShreyasZare committed:
DropRequests: Updated code to support comments in config. Added new "allowedLocalEndPoints" option in config and implemented it to allow requests coming only from listed local end points while dropping them from any other local end point.

ShreyasZare committed:
Failover: updated code to use InvariantCulture.

ShreyasZare committed:
GeoContinent: Updated code to support comments in config. Updated app to support ASN entries in record data.

ShreyasZare committed:
GeoCountry: Updated code to support comments in config. Implemented feature to support ASN entries.

ShreyasZare committed:
GeoDistance: Updated code to support comments in config.

ShreyasZare committed:
LogExporter: Updated code to use InvariantCulture. Code refactoring done for nullable related changes.

ShreyasZare committed:
QueryLogsMySql: Updated code to support comments in config. Nullable related changes done.

ShreyasZare committed:
QueryLogsSqlite: Updated code to support comments in config. Nullable related changes done. Updated code to use InvariantCulture.

ShreyasZare committed:
QueryLogsSqlServer: Updated code to support comments in config. Nullable related changes done.

ShreyasZare committed:
MispConnectorApp: removved the app since it is not feasible to be supported.

ShreyasZare committed:
updated solution file

jimstrang committed:
Refactored DNS log pagination to use LIMIT and OFFSET in QueryLogsMySql (#1317) (#1702)

* Refactor DNS log pagination to use LIMIT and OFFSET

Replaced the previous ROW_NUMBER()-based pagination with a simpler and more efficient approach using SQL LIMIT and OFFSET clauses. This change removes unnecessary subqueries and row number calculations, improving code clarity and query performance for paginated DNS log retrieval.

* Add sequential row numbers to DnsLogEntry results

Replace database ID with a calculated row number in DnsLogEntry objects, assigning sequential numbers based on the current page and sort order. This ensures each entry reflects its position in the paginated result set rather than its database ID.

* Improve pagination and query efficiency in DNS log queries

Refactored QueryLogsAsync to better handle invalid page numbers,
ensuring pageNumber is always within valid bounds. Database queries
are now only executed if there are entries to fetch, preventing
unnecessary queries and improving efficiency. Indented and grouped
parameter setup and result reading logic for clarity and correctness.

* Stricter validation for pageNumber in QueryLogsAsync

Changed pageNumber check to require values >= 1 instead of >= 0,
ensuring that page 0 cannot be used in log queries.

* Refactor log pagination to use SQL OFFSET/LIMIT

Replaced manual row range and ROW_NUMBER() logic with SQL OFFSET/LIMIT (SQLite) and OFFSET/FETCH (SQL Server) for more efficient pagination in QueryLogsAsync. Row numbers are now calculated in code, and queries are only executed if there are entries to fetch. This improves performance, readability, and maintainability.

* Refactor SQL param handling and reader loop logic

Refactored parameter addition to SQL commands to use explicit null checks, improving robustness and readability. Changed `@qclass` type from `ushort` to `short` for SQL Server. Updated `totalEntries` calculation to handle null results safely. Moved variable declarations and null checks inside the database reader loop, and restructured row number logic for clarity. No changes to query functionality.

ShreyasZare committed:
Query Logs Apps: fixed page number related bug introduced in PR #1702 that caused the last page navigation feature, which uses "-1" value, to stop working.

scj643 committed:
Initial PostgreSQL functionality. (#1600)

ShreyasZare committed:
QueryLogsMySql: Updated qtype to unsigned smallint. Code refactoring changes done.

ShreyasZare committed:
QueryLogsPostgres: Updated code to support comments in config. Updated BulkInsertLogsAsync() to handle truncated response. Updated qtype to INTEGER. Nullable related changes done. Updated QueryLogsAsync() to use LIMIT query for pagination. Code refactoring and minor changes done.

ShreyasZare committed:
QueryLogsSqlite: minor changes done.

ShreyasZare committed:
QueryLogsSqlServer: Updated qtype to INT. Other minor changes done.

daedaevibin committed:
Add Amber Theme for improved visual ergonomics and accessibility (#1810)

Co-authored-by: Dae Euhwa <daedaevibin@naver.com>

ShreyasZare committed:
webapp: moved dark mode and amber mode css code from main.css to independent css files. Updated html to include new css files.

ShreyasZare committed:
WebServiceAuthApi: Updated WriteSsoConfig() for minor changes.

zbalkan committed:
Added README for Apps (#1704)

* Added sample README

* Initial version of documentation of all default apps

Signed-off-by: Zafer Balkan <zafer@zaferbalkan.com>

* Updated README files

* Standardization

* Minor fixes

* Pushed README files when building

* Fix default values

* Fixed documentation error

* Config update

* Added GeoDB notes

* Fixed the mistake in IPv6 example

---------

Signed-off-by: Zafer Balkan <zafer@zaferbalkan.com>

ShreyasZare committed:
Merge branch 'app-docs-pr' into app-docs

ShreyasZare committed:
AdvancedForwarding: setting enableForwarding to false by default to avoid frequent support issues.

ShreyasZare committed:
app-docs: updated app documentation with minor fixes and additions.

ShreyasZare committed:
merged app-docs into develop

ShreyasZare committed:
fixed typo in config

ShreyasZare committed:
renamed postgres app

ShreyasZare committed:
updated app readme file to include database field values.

ShreyasZare committed:
updated publish profiles for dotnet 10.0

ShreyasZare committed:
minor formating changes

ShreyasZare committed:
uninstall.sh: updated script to ask before removing dotnet.

ShreyasZare committed:
DnsServer: Fixed issue with final response generation in ProcessCNAMEAsync() that caused issue with extra OPT in response causing exception.

ShreyasZare committed:
logs.js: fixed form reset issue that did not reset app class path to correct value.

ShreyasZare committed:
webapp: minor fixes.

ShreyasZare committed:
DnsWebService: updated ConfigureWebServiceRoutes() for metrics API.

ShreyasZare committed:
updated named.root

ShreyasZare committed:
WebServiceAuthApi: Updated SsoLoginFinalizeAsync() to improve logging messages.

ShreyasZare committed:
WebServiceDashboardApi: Added support for new metric API.

ShreyasZare committed:
inno: updated dotnet to v10.0.7 with code changes to simplify version changes in future.

ShreyasZare committed:
updated apidocs.

ShreyasZare committed:
updated solution file

ShreyasZare committed:
updated dockerfile for dotnet v10.0 upgrade

ShreyasZare committed:
minor update

ShreyasZare committed:
Updated dotnet to v10.0. App assembly version updated for release.

ShreyasZare committed:
app store json updated

ShreyasZare committed:
Dotnet updated to v10.0. Assembly version updated for release.

ShreyasZare committed:
inno: updated version to v15.0.

ShreyasZare committed:
updated build instructions

ShreyasZare committed:
updated changelog for v15.0 release.

ShreyasZare committed:
updated readme for v15.0 release.

ShreyasZare committed:
AuthManager: Updated ReadConfigFrom() to fix a bug that caused issue due to _ssoGroupMap being empty instead of null and another bug that caused failure to sync cluster api token.

ShreyasZare committed:
DnsWebService: Updated StartWebServiceAsync() to add additional scopes required for some SSO providers.

ShreyasZare committed:
WebServiceDashboardApi: updated GetMetricsText() to fix typo issue.

ShreyasZare committed:
assembly version updated to v15.0.1 for release

ShreyasZare committed:
inno: updated version to 15.0.1

ShreyasZare committed:
updated changelog for v15.0.1

ShreyasZare committed:
fixed readme formatting issues.

Wrong-Code committed:
Merge branch 'alpine-install' of https://github.com/Wrong-Code/DnsServer into alpine-install

# Conflicts:
#	DnsServerApp/install.sh

Update Alpine Linux support for Technitium DNS 15

Wrong-Code commented:
Hi! I’ve rebased this PR onto the latest upstream `master` and resolved the conflicts. The changes should now be up to date with the current codebase.

I tried to avoid bigger code reshuffles and obvious installation scripts cleanups, so that the PR has a chance to be adopted.

ShreyasZare commented:
Thanks for this. This PR still is quite tough to review due to all these changes in there. It would be best if you can create a new PR with just this install script and related changes.

I will try to test this soon before taking up any other major task.

Wrong-Code commented:
> Thanks for this. This PR still is quite tough to review due to all these changes in there. It would be best if you can create a new PR with just this install script and related changes.

It's because of my incompetence with Git. My fork was still based on the previous version of T. DNS, and when I rebased it to align the current version it incorporated all your commits (if you look closely, my changes indeed affect only your install.sh file, and I add another file, all the other changes are your commits).

If you still find the PR too busy, I can drop my fork and create another one. The script could be improved, but as I wrote I didn't want to overdo it.

Thanks!

ShreyasZare commented:
It would really be much better to have a new PR. Right now, its too tough to review.

Wrong-Code commented:
As requested, I have provided a rebuilt PR (#1889) with the same scope. I am closing this PR now.

ShreyasZare commented:
Thanks.
